packages/playground/src/pages/Dropin/Dropin.js

```diff
@@ -166,6 +166,23 @@ function handleRedirectResult() {
         });
     }
 
+    // Handle Amazon Pay redirect result
+    if (amazonCheckoutSessionId) {
+        window.amazonpay = checkout
+            .create('amazonpay', {
+                amazonCheckoutSessionId,
+                showOrderButton: false,
+                onSubmit: state => {
+                    makePayment(state.data).then(result => {
+                        handleFinalState(result.resultCode, dropin);
+                    });
+                }
```

Comment:
Is this intentional? Shouldn't it use the `onSubmit` event of the main instance?

Reply:
It is intentional. If I would use the one in the `onSubmit` in the main instance, it would try to call `component.setStatus` (with `component` being the Amazon Pay component), but I want to show the status in the Drop-in. In a real implementation this would most likely happen on another page without the Drop-in (just as a regular redirect flow) or at least it would show the result in another way. Also I tried to align this way of handling the redirect back with the one it's above, where we call the just `makeDetailsCall` directly.